src/index.js

```diff
@@ -10,6 +10,7 @@ const ownPkg = require('../package.json');
 const getLog = require('./lib/log');
 
 const pkg = JSON.parse(readFileSync('./package.json'));
+if (pkg.name === undefined) throw new Error(`"name" key is missing from your package.json`);
```

Comment:
Hi @gr2m, I updated my PR and moved the validation from **getRegistry** method in **npm.js** in the index.js after reading the package.json directly